### ISSUE TITLE
artifacts built in the wrong order by jitpack

### ISSUE THREAD

**Issue by akoehn** [dialogos-project/dialogos#191]
The dialogos-alexa plugin failed to build for both 2.1.0 and 2.1.1 because when jitpack tried to build it, dialogos was not available in the same version.   We need to make sure that everything is in place before building takes place.

I would have guessed that everything is nicely tagged in the git repos, but it seems like it was not.  @alexanderkoller, can you make an educated guess as to what went wrong?

I fixed for now it by removing the failed builds on jitpack. (BTW, 2.1.0 still fails, see [the build log](https://jitpack.io/com/github/dialogos-project/dialogos-alexa-plugin/2.1.0/build.log), but as it works with 2.1.1, I am not opening a new issue for that)

**Comment by timobaumann** [comment]
could this be related to the ordering of dependencies in the `build.gradle`s?

**Comment by akoehn** [comment]
Not really.  jitpack build everything on demand and as long as there is no demand for `dialogos-2.1.2` before there is a git tag `2.1.2` for dialogos, this problem should not arise.

If I understand correctly, the problem is triggered as follows:
 - some plugin `X` is tagged as `2.1.2`, which has a dependency `dialogos-2.1.2`
 - somebody wants to build something which depends on `X-2.1.2`
 - this request makes its way to jitpack
 - jitpack is unable to build  `X-2.1.2` because in the dialogos repository does not contain a tag `2.1.2` and the dependency can not be met.

**Comment by timobaumann** [comment]
then it's the ordering of tagging/building/pushing the tag to github in the build script?

**Comment by timobaumann** [comment]
my guess for what really happens (in the terminology from above): 
* somebody wants to build something that depends on `X-2.1.2`
* this request makes its way to jitpack
* jitpack needs to build first `X-2.1.2` and `dialogos-2.1.2` as a dependency
* building `dialogos-2.1.2` takes some time
* meanwhile, the build to `X-2.1.2` times out -- the build failed.
* jitpack never tries to re-build `X-2.1.2` because it believes it would also fail in the future.

maybe our make-release.py script should build some sort of dummy package that depends on `dialogos-2.1.2` from jitpack? that way we ensure that at least stuff on jitpack that directly depends on dialogos will build correctly.

**Comment by akoehn** [comment]
Interesting hypothesis, might be correct!

> maybe our make-release.py script should build some sort of dummy package that depends on dialogos-2.1.2 from jitpack? 

That should be possible with a simple wget to `https://jitpack.io/com/github/dialogos-project/dialogos/X.X.X/dialogos-X.X.X.pom`.

**Comment by timobaumann** [comment]
hard to test though, because jitpack says "Tag builds older than 7 days can't be deleted."

**Comment by timobaumann** [comment]
Arne, can you add the wget to the release-script so that we can monitor whether this works in the future?

**Comment by akoehn** [comment]
Not until in about a week or so :-/
